### ISSUE TITLE
Git Bash: gpg --full-generate-key hangs. Can't generate key

### ISSUE THREAD

**Issue by kienerj** [git-for-windows/git#3768]
- [ ] I was not able to find an [open](https://github.com/git-for-windows/git/issues?q=is%3Aopen) or [closed](https://github.com/git-for-windows/git/issues?q=is%3Aclosed) issue matching what I'm seeing

### Setup

 - Which version of Git for Windows are you using? Is it 32-bit or 64-bit?
 - Windows 10
 - Git-2.35.1.2-64-bit
  
```
$ git --version --build-options

** $ git --version --build-options
git version 2.35.1.windows.2
cpu: x86_64
built from commit: 5437f0fd368c7faf1a0b5e1fef048232c1f2a3e6
sizeof-long: 4
sizeof-size_t: 8
shell-path: /bin/sh
feature: fsmonitor--daemon
 **
```

 - Which version of Windows are you running? Vista, 7, 8, 10? Is it 32-bit or 64-bit?

```
$ cmd.exe /c ver

** Microsoft Windows [Version 10.0.18363.2158] **
```

 - What options did you set as part of the installation? Or did you choose the
   defaults?

```

**
Editor Option: Notepad++
Custom Editor Path: 
Default Branch Option:  
Path Option: Cmd
SSH Option: OpenSSH
Tortoise Option: false
CURL Option: OpenSSL
CRLF Option: CRLFAlways
Bash Terminal Option: MinTTY
Git Pull Behavior Option: Merge
Use Credential Manager: Enabled
Performance Tweaks FSCache: Enabled
Enable Symlinks: Disabled
Enable Pseudo Console Support: Disabled
Enable FSMonitor: Disabled
 **
```

### Details

 - Which terminal/shell are you running Git from? e.g Bash/CMD/PowerShell/other

** Git Bash **

 - What commands did you run to trigger this issue? If you can provide a
   [Minimal, Complete, and Verifiable example](http://stackoverflow.com/help/mcve)
   this will help us understand the issue.

```
** gpg --full-generate-key --expert **
```
 - What did you expect to occur after running these commands?

** The key generation programm to start and asking for options **

 - What actually happened instead?
 
Running the command (with or without expert flag) just hangs and nothing happens. 

![image](https://user-images.githubusercontent.com/1567309/161698675-2757d3f2-ceb2-418c-ae11-6e333e145dfa.png)

I can then terminate with ctrl+c but I can't generate a new key.

**Comment by dscho** [comment]
Does the same happen if you call that command in [MSYS2](https://msys2.github.io/)?

**Comment by kienerj** [comment]
> Does the same happen if you call that command in [MSYS2](https://msys2.github.io/)?

Yes. I downloaded it, perfromed the updated and entered the same gpg command with the same effect that nothing appears to happen.

**Comment by kienerj** [comment]
There seem to be many issues at play why I can't get what I want. the core issue was that I enabled automatic git key signing which means adding below options in gpg.conf. Removing them and it works again.

```
batch
no-tty
```

On top of that i encounter other issues, namely not being able to edit subkeys. Cause being that have the master key offline /on a different machine so no subkey can be changed on this specific machine.

Closing as not related to git for windows.
